src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.spec.ts

```diff
@@ -211,6 +211,21 @@ describe('DotListingDataTableComponent', () => {
         expect(url).toEqual(paginatorService.url);
     });
 
+    it('should not call paginatorService when firstPageData comes', () => {
+        spyOn(paginatorService, 'getWithOffset');
+        comp.firstPageData = items;
+        comp.columns = columns;
+        comp.url = url;
+
+        comp.ngOnChanges({
```

Comment:
We should not use this... `fixture.detectChanges` instead, this approach was a rookie mistake.